_posts/en/newsletters/2023-09-13-newsletter.md

```diff
@@ -0,0 +1,186 @@
+  set of messages for handling legacy HTLC forwarding while also
+  offering upgraded messages to support PTLC forwarding, creating two
+  different paths that will need to be maintained concurrently until
+  HTLCs are phased out.  If some implementations add experimental PTLCs
+  support before the messages are standardized, then implementations
+  might even be required to support three or more different protocols
+  simultaneously, to the disadvantage of all.
+
+    Sander's summary had not received any comments as of this writing.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review
+Club][] meeting, highlighting some of the important questions and
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[Transport abstraction][review club 28165] is a recently-merged PR by Pieter Wuille (sipa)
+that introduces a _transport_ abstraction (interface class). Concrete
+derivations of this class convert a (per-peer) connection's (already
+serialized) send and receive messages to and from wire format. This
+can be thought of as implementing a deeper level of serialization and
+deserialization. These classes do not do the actual sending and receiving.
+
+The PR derives two concrete classes from the `Transport` class,
+`V1Transport` (what we have today) and `V2Transport` (encrypted on the
+wire). This PR is part of the
+[BIP324][topic v2 p2p transport]
+_Version 2 P2P Encrypted Transport Protocol_ [project][v2 p2p tracking pr].
+
+{% include functions/details-list.md
+  q0="What is the distinction between [*net*][net] and
+      [*net_processing*][net_processing]?"
+  a0="*Net* is at the bottom of the networking stack and handles
+       low-level communication between peers, while *net_processing*
+       builds on top of the *net* layer and handles the processing
+       and validation of messages from *net* layer."
+  a0link="https://bitcoincore.reviews/28165#l-22"
+
+  q1="More concretely, name examples of classes or functions that
+      we'd associate with *net_processing*, and, in contrast,
+      with *net*?"
+  a1="*net_processing*: `PeerManager`, `ProcessMessage`.
+      *net*: `CNode`, `ReceiveMsgBytes`, `CConnMan`."
+  a1link="https://bitcoincore.reviews/28165#l-25"
+
+  q2="Does BIP324 require changes to the *net* layer, the
+      *net_processing* layer, or both? Does it affect policy
+      or consensus?"
+  a2="These changes are only at the *net* layer; they don't affect consensus."
+  a2link="https://bitcoincore.reviews/28165#l-37"
+
+  q3="What are examples of implementation bugs that could result in
+      this PR being an (accidental) consensus change?"
+  a3="A bug that restricts the maximum message size less to than
+      4MB, resulting in a blockweight soft fork; a bug in block
+      deserialization that causes the node to reject a consensus-valid
+      block, also a soft fork."
+  a3link="https://bitcoincore.reviews/28165#l-45"
+
```

Comment:
Although instagibbs calls the <4MB theoretical bug a soft fork, neither glozow nor anyone else I see calls the generalized deserialization problem a soft fork.  I'm not sure everything in that class of failures would be a soft fork; e.g., a non-deterministic deserialization failure (kinda similar to the BIP50 failure) wouldn't be a soft fork; more relevant, the type of platform-inconsistent serialization/deserialization behavior that BIP66 surreptitiously fixed isn't something I'd have described as a soft fork if it had been triggered and caused a consensus failure.

In other words, I think we should drop the phrase ", also a soft fork" from the end of this paragraph.  (But I don't have strong feelings about it.)

Reply:
So, to be designated a soft fork, the behavior change in question must be intentional?

But I'm fine with dropping both mentions of "soft fork" if it's at all confusing, I think the answer will still be understandable and accurate.

Reply:
@LarryRuane 

> So, to be designated a soft fork, the behavior change in question must be intentional?

Maybe?  But the point I'm trying to make is that, to be designated a soft fork, the behavior needs to be consistent.  If the behavior between nodes is inconsistent, it's a _consensus failure_.  We might eventually resolve a consensus failure using a soft fork (e.g., BIP50 resolved the BDB locks inconsistency with a soft fork that temporarily reduced the maximum block size below the level where the maximum lock count was reached, and BIP66 resolved the DER decoding inconsistency with a stricter decoding rule) but my point is that a bug in the (de)serialization code wouldn't necessarily produce a soft fork because it wouldn't necessarily occur consistently between full nodes, so it could cause a consensus failure (and consensus failures are not soft forks, at least in my personal terms sheet).

Reply:
That's an excellent way to look at it, thanks, @harding!